File: ios/Anemone SDK/general/extension/DateRange+Json.swift
//
//  DateRange+Json.swift
//  Anemone SDK
//
//  Created by Angel Luis Garcia on 1/2/21.
//  Copyright © 2021 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public extension TimeZoneDateRange {
    static func from(json: JSONObject) throws -> TimeZoneDateRange {
        TimeZoneDateRange(
            from: try json.getDate("from"),
            to: try json.getDate("to"),
            timeZone: TimeZone(identifier: json.optString("timezone") ?? "") ?? TimeZone.current
        )
    }

    func toJson() -> JSONObject {
        let json = JSONObject()
        json.setDate("from", from)
        json.setDate("to", to)
        json.setString("timezone", timeZone.identifier)
        return json
    }
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/general/extension/DateRange+Json.kt
package com.anemonesdk.general.extension

import com.anemonesdk.general.json.JSONObject
import com.playtomic.foundation.extension.tryOrNull
import com.playtomic.foundation.model.TimeZoneDateRange
import java.util.*

fun TimeZoneDateRange.Companion.from(json: JSONObject): TimeZoneDateRange =
    TimeZoneDateRange(
        from = json.getDate("from"),
        to = json.getDate("to"),
        timeZone = tryOrNull { TimeZone.getTimeZone(json.optString("timezone") ?: "") }
            ?: TimeZone.getDefault()
    )

fun TimeZoneDateRange.toJson(): JSONObject {
    val json = JSONObject()
    json.setDate("from", from)
    json.setDate("to", to)
    json.setString("timezone", timeZone.id)
    return json
}
